tests/lib/ifaces/linux_bridge_iface_test.py

```diff
@@ -0,0 +1,420 @@
+            deepcopy(SLAVE2_PORT_CONFIG)
+        ]
+        des_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = [
+            {Port.NAME: SLAVE2_IFACE_NAME},
+        ]
+
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+        des_iface.merge(cur_iface)
+
+        assert des_iface.to_dict() == expected_iface_info
+
+    def test_merged_with_desire_remove_all_ports(self):
+        cur_iface_info = gen_bridge_iface_info()
+        des_iface_info = gen_bridge_iface_info()
+        expected_iface_info = gen_bridge_iface_info()
+        expected_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = []
+        des_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = []
+
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+        des_iface.merge(cur_iface)
+
+        assert des_iface.to_dict() == expected_iface_info
+
+    def test_merge_with_desire_port_not_defined(self):
+        cur_iface_info = gen_bridge_iface_info()
+        des_iface_info = gen_bridge_iface_info()
+        expected_iface_info = gen_bridge_iface_info()
+        des_iface_info[LB.CONFIG_SUBTREE].pop(LB.PORT_SUBTREE)
+
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+        des_iface.merge(cur_iface)
+
+        assert des_iface.to_dict() == expected_iface_info
+
+    def test_desire_port_name_partial_merge_from_current_with_vlan_filter(
+        self,
+    ):
+        cur_iface_info = gen_bridge_iface_info_with_vlan_filter()
+        des_iface_info = gen_bridge_iface_info_with_vlan_filter()
+        expected_port_config = deepcopy(SLAVE2_PORT_CONFIG)
+        expected_port_config[Port.VLAN_SUBTREE] = SLAVE2_VLAN_CONFIG_TRUNK
+        expected_iface_info = gen_bridge_iface_info()
+        expected_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = [
+            expected_port_config
+        ]
+        des_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = [
+            {Port.NAME: SLAVE2_IFACE_NAME},
+        ]
+
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+        des_iface.merge(cur_iface)
+
+        assert des_iface.to_dict() == expected_iface_info
+
+    def test_fix_access_mode_with_enable_native(self):
+        cur_iface_info = gen_bridge_iface_info_with_vlan_filter()
+        des_iface_info = gen_bridge_iface_info_with_vlan_filter()
+        expected_iface_info = gen_bridge_iface_info_with_vlan_filter()
+        cur_vlan_conf = cur_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE][0][
+            Port.VLAN_SUBTREE
+        ]
+        cur_vlan_conf[Vlan.ENABLE_NATIVE] = True
+        cur_vlan_conf[Vlan.MODE] = Vlan.Mode.TRUNK
+
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+        des_iface.merge(cur_iface)
+        des_iface.pre_edit_validation_and_cleanup()
```

Comment:
In the refactoring of the `bridgelib` - which in some ways makes the test code more readable - you've removed the ability of injecting the vlan configuration from the test, using instead some constants defined in the bridgelib. 

While it makes the code more concise, I think it impacts the readability of the test code as a whole - e.g. imho, it would be a lot more readable if the original port vlan configuration was defined in the test ifself.

Maybe just getting rid of the `gen_bridge_iface_info_with_vlan_filter()` and adding the port vlan config to the current state **in** each test would be enough... (but I get the feeling this is part of what you intend to abstract away.)

Reply:
You still can change the returned dict from `gen_bridge_iface_info_with_vlan_filter()` for njecting the vlan configuration from the test.

The existence of `gen_bridge_iface_info_with_vlan_filter` is to eliminate the duplicate code, no fancy big picture plan involved.